turbopack/crates/turbo-persistence/src/static_sorted_file.rs

```diff
@@ -339,6 +339,19 @@ impl StaticSortedFile {
 
     /// Reads a block from the file.
     fn read_block(&self, block_index: u16, compression_dictionary: &[u8]) -> Result<ArcSlice<u8>> {
+        let (uncompressed_length, block) = self.get_compressed_block(block_index)?;
+
+        let buffer = Arc::new_zeroed_slice(uncompressed_length as usize);
+        // Safety: MaybeUninit<u8> can be safely transmuted to u8.
+        let mut buffer = unsafe { transmute::<Arc<[MaybeUninit<u8>]>, Arc<[u8]>>(buffer) };
```

Comment:
While in practice this is true, it's not guaranteed to be true. Non-`repr(transparent)` container types can have different representations even if their contained type has the same representation.

`assume_init` for `Box` uses `into_raw`/`from_raw`: https://github.com/rust-lang/rust/blob/4b55fe199cfe9c710555a5af7f2a49491ad38254/library/alloc/src/boxed.rs#L992

`Arc` has the same APIs, so you can use that instead of transmuting the entire container type.